com.unity.ml-agents/Runtime/Sensors/RayPerceptionSensorComponentBase.cs

```diff
@@ -15,12 +15,12 @@ public abstract class RayPerceptionSensorComponentBase : SensorComponent
 
         /// <summary>
         /// The name of the Sensor that this component wraps.
+        /// Note that changing this at runtime does not affect how the Agent sorts the sensors.
         /// </summary>
         public string sensorName
         {
-            get => m_SensorName;
-            // Restrict the access on the name, since changing it a runtime doesn't re-sort the Agent sensors.
-            internal set => m_SensorName = value;
+            get { return m_SensorName; }
+            set { m_SensorName = value; UpdateSensor(); }
```

Comment:
In `CameraSensorComponent.cs` changing the sensorName does not trigger a UpdateSensor. Is this normal?

Reply:
I should probably remove this call here, since the name isn't propagated to the sensor by UpdateSensor anyway.